### PR TITLE
Add new session verification details dialog

### PR DESCRIPTION
**Pull request by jryans**
This gives more info on the session you're about to verify, including device
name and ID.

2020-01-27 at 23 17

Fixes https://github.com/vector-im/riot-web/issues/11977

### CODE REVIEW

File: src/components/views/dialogs/NewSessionReviewDialog.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+limitations under the License.
+*/
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import * as sdk from "../../../index";
+import { _t } from '../../../languageHandler';
+import Modal from "../../../Modal";
+
+export default class NewSessionReviewDialog extends React.PureComponent {
```
Comment (turt2live) [inline src/components/views/dialogs/NewSessionReviewDialog.js]: `@replaceableComponent` please

File: src/components/views/dialogs/NewSessionReviewDialog.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        onFinished: PropTypes.func.isRequired,
+    }
+
+    onCancelClick = () => {
+        this.props.onFinished(false);
+    }
+
+    onContinueClick = () => {
+        const DeviceVerifyDialog =
+            sdk.getComponent('views.dialogs.DeviceVerifyDialog');
```
Comment (turt2live) [inline src/components/views/dialogs/NewSessionReviewDialog.js]: to make the transition easier, I'd recommend starting to just `import` the components you need assuming it doesn't burst into flames when you do so (there's a couple components that do).

File: src/components/views/elements/DialogButtons.js [hunk 3]
```diff
@@ -83,7 +83,7 @@ export default createReactClass({
                 // primary in the DOM so will get form submissions unless we make it not a submit.
                 type="button"
                 onClick={this._onCancelClick}
-		className={this.props.cancelButtonClass}
+                className={this.props.cancelButtonClass}
```
Comment (turt2live) [inline src/components/views/elements/DialogButtons.js]: how in the...

File: src/components/views/dialogs/NewSessionReviewDialog.js (hunk 2 above)
Comment (turt2live) [reply src/components/views/dialogs/NewSessionReviewDialog.js]: applies to whole PR

File: src/components/views/elements/DialogButtons.js (hunk 3 above)
Comment (turt2live) [reply src/components/views/elements/DialogButtons.js]: ah, https://github.com/matrix-org/matrix-react-sdk/pull/3954 makes sense now.

File: src/components/views/dialogs/NewSessionReviewDialog.js (hunk 1 above)
Comment (jryans) [reply src/components/views/dialogs/NewSessionReviewDialog.js]: Will do, thanks!

File: src/components/views/dialogs/NewSessionReviewDialog.js (hunk 2 above)
Comment (jryans) [reply src/components/views/dialogs/NewSessionReviewDialog.js]: Done, thanks!

**Pull request merged**